public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -96,16 +98,24 @@ export class QueryRows extends PureComponent<Props> {
     );
   };
 
-  onChangeDataSource = (settings: DataSourceInstanceSettings, index: number) => {
+  onChangeDataSource = async (settings: DataSourceInstanceSettings, index: number) => {
     const { queries, onQueriesChange } = this.props;
+    const dataSource = await getDataSourceSrv().get(settings.uid);
```

Comment:
I'm not quite sure what will happen if this request fails, but I think it might halt script execution. Best to wrap this in a `try...catch` block?

Reply:
Agree, but I think I will revert the `ds.getDefaultQuery?.` functionality and open an issue for it because I saw that the logic for adding a new query is located in a reducer. So we probably need to this this there as well and it is probably better to do that in both places in a separate PR.

Reply:
I opened this issue: https://github.com/grafana/grafana/issues/61280

Reply:
Do you want to give it a go or should I @gillesdemey?